src/coreclr/jit/fgopt.cpp

```diff
@@ -5309,6 +5309,24 @@ PhaseStatus Compiler::fgDfsBlocksAndRemove()
             m_dfsTree = fgComputeDfs();
         }
 
+#ifdef DEBUG
+        // Did we actually remove all the blocks we said we were going to?
+        if (verbose)
+        {
+            if (m_dfsTree->GetPostOrderCount() != fgBBcount)
+            {
+                printf("%u unreachable blocks were not removed:\n", fgBBcount - m_dfsTree->GetPostOrderCount());
+                for (BasicBlock* block : Blocks())
+                {
+                    if (!m_dfsTree->Contains(block))
+                    {
+                        printf("  " FMT_BB "\n", block->bbNum);
+                    }
+                }
+            }
+        }
+#endif // DEBUG
```

Comment:
Luckily the blocks are still removed in the "graph" sense -- they don't have any preds/succs left anymore, so they don't interact with reachable blocks in any way. I.e. asserts like these are ok:
https://github.com/dotnet/runtime/blob/b60a541554cc075b903a4d218456ef8588092a8c/src/coreclr/jit/morph.cpp#L13833-L13835

But yeah, we still have this annoying case to check for when iterating the block list.